apps/ui/src/modules/marketplace/components/mintPanel/mintContent.tsx

```diff
@@ -11,19 +11,47 @@ import {
 } from '@chakra-ui/react';
 import { useState } from 'react';
 import { AddIcon, MinusIcon } from '@chakra-ui/icons';
-import BlueEthIcon from '@/assets/marketplace/blue-eth.svg';
+import type { AssetInfo, BN } from 'fuels';
+import UnknownAsset from '@/assets/unknown-asset.png';
+import { convertToUsd } from '@/utils/convertToUsd';
 
-const MintContent = (props: {
+type MintContentProps = {
   title: string;
   description: string;
   progress: number;
   maxSupply: number;
-  maxPerWallet: number;
-  priceEth: number;
-  priceUsd: number;
-}) => {
+  // maxPerWallet: number;
+  tokenPrice: BN;
+  isMinting: boolean;
+  asset: AssetInfo | null | undefined;
+  onMint: (quantity: number) => void;
+};
+
+const MintContent = ({
+  title,
+  description,
+  progress,
+  maxSupply,
+  // maxPerWallet,
+  tokenPrice,
+  isMinting,
+  asset,
+  onMint,
+}: MintContentProps) => {
   const [quantity, setQuantity] = useState(1);
-  const progress = (props.progress / props.maxSupply) * 100;
+  const progressPercentage = (progress / maxSupply) * 100;
+
+  const mintPrice = tokenPrice.mul(quantity).formatUnits(asset?.decimals ?? 0);
+  const usdPrice = convertToUsd(
+    tokenPrice.mul(quantity),
+    asset?.decimals ?? 0,
+    asset?.rate ?? 0
+  ).formatted;
```

Comment:
add `useMemo` for avoid unnecessary calculations

```suggestion
  const progressPercentage = useMemo(() => (progress / maxSupply) * 100, [progress, maxSupply])

  const mintPrice = useMemo(
  () => tokenPrice.mul(quantity).formatUnits(asset?.decimals ?? 0),
  [tokenPrice, quantity, asset?.decimals]
);

const usdPrice = useMemo(
  () =>
    convertToUsd(
      tokenPrice.mul(quantity),
      asset?.decimals ?? 0,
      asset?.rate ?? 0
    ).formatted,
  [tokenPrice, quantity, asset?.decimals, asset?.rate]
);
```